lib/concurrent-ruby/concurrent/executor/java_single_thread_executor.rb

```diff
@@ -17,12 +17,13 @@ def initialize(opts = {})
       end
 
       private
-      
+
       def ns_initialize(opts)
-        @executor = java.util.concurrent.Executors.newSingleThreadExecutor
+        @executor = java.util.concurrent.Executors.newSingleThreadExecutor(
```

Comment:
@pitr-ch this was the missing piece.
Locally my `rake spec` now completes & doesn't hang.
I had to move **DaemonThreadFactory** to the parent class.

Reply:
Thank you for investigating and the fix!